axelor-base/src/main/java/com/axelor/apps/base/service/imports/importer/FactoryImporter.java

```
@@ -26,28 +26,23 @@
 public class FactoryImporter {
 
   public Importer createImporter(ImportConfiguration importConfiguration) {
-
-    Importer importer;
-
-    if (importConfiguration.getTypeSelect().equals(ImportConfigurationRepository.TYPE_XML)) {
-      importer = Beans.get(ImporterXML.class);
-    } else {
-      importer = Beans.get(ImporterCSV.class);
-    }
-
+    Importer importer = getImporter(importConfiguration.getTypeSelect());
     return importer.init(importConfiguration);
   }
 
   public Importer createImporter(ImportConfiguration importConfiguration, File workspace) {
+    Importer importer = getImporter(importConfiguration.getTypeSelect());
+    return importer.init(importConfiguration, workspace);
+  }
 
-    Importer importer;
-
-    if (importConfiguration.getTypeSelect().equals(ImportConfigurationRepository.TYPE_XML)) {
-      importer = Beans.get(ImporterXML.class);
-    } else {
-      importer = Beans.get(ImporterCSV.class);
+  protected Importer getImporter(String typeSelect) {
+    switch (typeSelect) {
+      case ImportConfigurationRepository.TYPE_XML:
+        return Beans.get(ImporterXML.class);
+      case ImportConfigurationRepository.TYPE_CSV:
+        return Beans.get(ImporterCSV.class);
+      default:
+        return null;
```

in this case, we should throw an exception